pkg/build/builder/source.go

```diff
@@ -229,6 +225,14 @@ func extractGitSource(gitClient GitClient, gitSource *api.GitBuildSource, revisi
 		if err := gitClient.SubmoduleUpdate(dir, true, true); err != nil {
 			return true, err
 		}
+		if err := gitClient.CloneWithOptions(dir, gitSource.URI, cloneOpts); err != nil {
```

Comment:
this is reversing the order we do these operations in... that doesn't seem right